module-apps/apps-common/widgets/DateSetSpinner.cpp

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "DateSetSpinner.hpp"
+
+#include "BellSideListItemStyle.hpp"
+#include <FontManager.hpp>
+#include <RawFont.hpp>
+#include <apps-common/widgets/BellBaseLayout.hpp>
+#include <gui/widgets/ImageBox.hpp>
+#include <gui/widgets/text/Label.hpp>
+#include <time/time_locale.hpp>
+
+namespace gui
+{
+    namespace date_set_spinner
+    {
+        namespace focus
+        {
+            constexpr auto size = 4U;
+        } // namespace focus
+
+        constexpr auto focusFontName    = style::window::font::large;
+        constexpr auto noFocusFontName  = style::window::font::largelight;
+        constexpr std::uint8_t step     = 1;
+        constexpr std::uint8_t dayMin   = 1;
+        constexpr std::uint8_t dayMax   = 31;
+        constexpr std::uint8_t monthMin = 1;
+        constexpr std::uint8_t monthMax = 12;
+
+    } // namespace date_set_spinner
+
+    using namespace date_set_spinner;
+
+    DateSetSpinner::DateSetSpinner(Item *parent, VBox *titleBox, Length x, Length y, Length w, Length h)
+        : HBox(parent, x, y, w, h)
+    {
+        setAlignment(Alignment(Alignment::Horizontal::Center, Alignment::Vertical::Center));
+        setEdges(RectangleEdge::None);
+
+        title = new TextFixedSize(titleBox);
+        title->setMaximumSize(style::bell_base_layout::w, style::bell_base_layout::outer_layouts_h);
+        title->setFont(style::bell_sidelist_item::title_font);
```

Comment:
Is this a generic widget that can be used both for Pure/Harmony? If yes, then these definitions cannot be used as they inject harmony-related dependencies. I can see two solutions to this problem:
* move this module to the bell product directory
* make it truly generic(remove harmony dependencies)
Whatever you choose, I'm ok with that.

Reply:
I did choose option number 2. I added 2 setters and these are set in DateSetListItem. I did remove not needed :
#include "BellSideListItemStyle.hpp"
and also
#include <apps-common/widgets/BellBaseLayout.hpp>